src/main/webapp/mentors.js

```diff
@@ -122,3 +122,32 @@ function createProjectsDiv(projectIDs) {
   }
   return projectsDiv;
 }
+
+function submitForm() {
+  const inputUrl = document.getElementById("inputRepo").value;
+  const encodedBody = new URLSearchParams();
+  encodedBody.append('inputRepo', inputUrl);
+  let url = new URL(
+      '/mentors', window.location.protocol + '//' + window.location.hostname);
+  const fetchRequest = fetch((url), {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: encodedBody,
+  });
+  
+  fetchRequest.then((response) => response.json()).then((response) => {
+    if (response == 'success') {
+      window.location.href = '/dashboard.html';
+    }
+    else {
+      const errorContainer = document.getElementById('error-message-container');
+      errorContainer.innerHTML = response;
+    }
+  }).catch((errorResponse) => {
+    console.error(
+        `Error ${errorResponse.statusCode}: ${errorResponse.error}`);
+      alert(errorResponse.userMessage);
+  });
```

Comment:
`errorResponse.userMessage` is only defined for an error when using the fetch error handler. I suspect the intention was to use the fetch error handler. If so you need to wrap the fetch request:
```javascript
standardizeFetchErrors(fetchRequest, 'Fetch failed message here', 'Server error message here').then((response) => [...]);

Reply:
I am def going to use it! I changed my js file into a module, added import statements, etc. in another pull request tho, so I will not wrap this function until that PR is merged (so I don't do the exact same thing in two PRs)

Reply:
Update: I did it anyways, I will just fix the merge conflict later. :/